timeboost/s3_storage.go

```diff
@@ -0,0 +1,196 @@
+package timeboost
+
+import (
+	"bytes"
+	"context"
+	"encoding/csv"
+
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/util"
+	"github.com/offchainlabs/nitro/util/s3client"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	"github.com/spf13/pflag"
+)
+
+type S3StorageServiceConfig struct {
+	Enable         bool          `koanf:"enable"`
+	AccessKey      string        `koanf:"access-key"`
+	Bucket         string        `koanf:"bucket"`
+	ObjectPrefix   string        `koanf:"object-prefix"`
+	Region         string        `koanf:"region"`
+	SecretKey      string        `koanf:"secret-key"`
+	UploadInterval time.Duration `koanf:"upload-interval"`
+	MaxBatchSize   int           `koanf:"max-batch-size"`
+	MaxDbRows      int           `koanf:"max-db-rows"`
+}
+
+func (c *S3StorageServiceConfig) Validate() error {
+	if !c.Enable {
+		return nil
+	}
+	if c.MaxBatchSize < 0 {
+		return fmt.Errorf("invalid max-batch-size value for auctioneer's s3-storage config, it should be non-negative, got: %d", c.MaxBatchSize)
+	}
+	if c.MaxDbRows < 0 {
+		return fmt.Errorf("invalid max-db-rows value for auctioneer's s3-storage config, it should be non-negative, got: %d", c.MaxDbRows)
+	}
+	return nil
+}
+
+var DefaultS3StorageServiceConfig = S3StorageServiceConfig{
+	Enable:         false,
+	UploadInterval: time.Minute, // is this the right default value?
```

Comment:
Let's set it to 15 minutes. This results in fewer S3 put requests per day for the auctioneer and fewer get requests for users of the data, resulting in less cost, while maintaining reasonably fresh data.